arviz/data/io_dict.py

```diff
@@ -69,11 +69,15 @@ def __init__(
         self.attrs.pop("created_at", None)
         self.attrs.pop("arviz_version", None)
 
-    @requires("posterior")
+    def _init_dict(self, attr_name):
+        dict_or_none = getattr(self, attr_name, None)
```

Comment:
should this return empty dict instead of None?

Reply:
yeah, much better, it directly takes care about the if else below, thanks!

Reply:
now I realized the if else is what does the actual work, for now all attrs are set, but it's nice to have this here in case we extended io_dict.

Reply:
Remember to update the block below too

Reply:
The block below needs to check for None. This first one catches attributes that have not been set and the one below catches attributes set to None